src/portable/st/stm32_fsdev/dcd_stm32_fsdev.c

```diff
@@ -656,7 +660,14 @@ void dcd_int_handler(uint8_t rhport) {
 
   (void) rhport;
 
-  uint32_t int_status = USB->ISTR;
+  // Only handle interrupts which are triggered and currently active
+  uint32_t int_status = USB->ISTR & USB->CNTR;
```

Comment:
Huh? I don't get why this matters. Sounds ridicules to me to use:
```
uint32_t int_status = USB->ISTR;
int_status &= USB->CNTR;
```

I don't get why a compiler should throw a warning here.

Reply:
> Huh? I don't get why this matters. Sounds ridicules to me to use:

That's definitely is **not non-sense**

> I don't get why a compiler should throw a warning here, the order shouldn't be defined by C either and the compiler is free to reorder instructions anyway. The order which register is read first here also don't matter.

The order of register access is important in some cases, register read has side effects such as clear/set bits.

![image](https://github.com/hathach/tinyusb/assets/4375114/2822a8c2-b679-4e52-a0f4-b668ef5444f0)

Reply:
Would be really nice if you can just provide an example where this matter. Showing that operations can have side effects don't add any value about this discussion.

Reply:
Hmm, I guess you are right and it can cause issues for some very rare things. Like timing related stings in a communication mesh. For example when different SPI secondaries need to be accessed in a very specific order.